File: Easy/771 - Jewels and Stones/771Soln-B.rb
# frozen_string_literal: true

# @param {String} j
# @param {String} s
# @return {Integer}
def num_jewels_in_stones(j, s)
  count = 0
  for jewel in j.split('') do
    for stone in s.split('') do
      jewel == stone ? count += 1 : count
    end
  end

  count
end

File: Easy/771 - Jewels and Stones/771Soln-B.js
'use strict'

/**
 * @param {string} J
 * @param {string} S
 * @return {number}
 */
const numJewelsInStones = function(J, S) {
    let count = 0
    for (let i = 0; i < J.length; i++) {
        for (let j = 0; j < S.length; j++) {
            if (J[i] === S[j]) { count++ }
        }
    }

    return count
}
